packages/aws-cdk/lib/api/logs/logs-monitor.ts

```diff
@@ -0,0 +1,255 @@
+    this.printer = options.printer ?? new EventPrinter();
+
+    this.scheduleNextTick();
+  }
+
+  // allows the ability to "pause" the monitor. The primary
+  // use case for this is when we are in the middle of performing a deployment
+  // and don't want to interweave all the logs together
+  public activate(): void {
+    this.active = true;
+  }
+
+  public deactivate(): void {
+    this.active = false;
+  }
+
+  /**
+   * Adds CloudWatch log groups to read log events from.
+   * Since we could be watching multiple stacks that deploy to
+   * multiple AWS accounts, we need to store a list of log groups
+   * per account along with the SDK object that has access to read from
+   * that AWS account.
+   */
+  public addLogGroups(account: string, config: LogGroupConfig): void {
+    const logGroup = this.logGroups.get(account);
+    if (logGroup) {
+      config.groups.forEach(group => {
+        logGroup.groups.add(group);
+        this.groupStartTime.set(group, this.startTime);
+      });
+    } else {
+      this.logGroups.set(account, config);
+    }
+  }
+
+  /**
+   * reset the list of log groups that are being monitored
+   * along with the startTime.
+   *
+   * This should only be triggered prior to a CFN deployment
+   */
+  public resetLogGroups(): void {
+    this.startTime = Date.now();
+    this.logGroups.clear();
+  }
+
+  private scheduleNextTick(): void {
+    setTimeout(() => void(this.tick()), SLEEP);
+  }
+
+  private async tick(): Promise<void> {
+    if (!this.active) {
+      this.scheduleNextTick();
+      return;
+    }
+    try {
+      const events = flatten(await this.readNewEvents());
+      events.forEach(event => {
+        this.printer.print(event);
+      });
+    } catch (e) {
+      error('Error occurred while monitoring logs: %s', e);
+    }
+    this.scheduleNextTick();
+  }
+
+  /**
+   * Reads all new log events from a set of CloudWatch Log Groups
+   * in parallel
+   */
+  private async readNewEvents(): Promise<Array<Array<CloudWatchLogEvent>>> {
+    const promises: Array<Promise<Array<CloudWatchLogEvent>>> = [];
+    for (const config of this.logGroups.values()) {
+      const sdk = config.sdk;
+      for (const group of config.groups) {
+        promises.push(this.readEventsFromLogGroup(sdk, group));
+      }
+    }
+    return Promise.all(promises);
+  }
+
+  /**
+   * Reads all new log events from a CloudWatch Log Group
+   * starting when the hotswap was triggered
+   *
+   * Only prints out events that have not been printed already
+   */
+  private async readEventsFromLogGroup(sdk: ISDK, logGroupName: string): Promise<CloudWatchLogEvent[]> {
+    const events: CloudWatchLogEvent[] = [];
+
+    // log events from some service are ingested faster than others
+    // so we need to track the start/end time for each log group individually
+    // to make sure that we process all events from each log group
+    const startTime = this.groupStartTime.get(logGroupName) ?? this.startTime;
```

Comment:
I noticed an issue with using a universal `startTime` so this now tracks a `startTime` per logGroup.